From 26a4798723d74c4b9049946fde35e1c2332fc915 Mon Sep 17 00:00:00 2001
From: c4pp4
Date: Sun, 25 Oct 2026 03:52:31 +0100
Subject: [PATCH 1/1] Do not read thumbnails from Top Sites

Chromium does not store the page thumbnails in the Top Sites database
anymore, so the query of the missing thumbnails table fails and no
bookmarks are found. Disable the thumbnails, the bookmarks are shown
with the default icon.

Signed-off-by: c4pp4
---
 src/unity_chromiumbookmarks_daemon.py | 50 +++++++++++++--------------
 1 file changed, 25 insertions(+), 25 deletions(-)

diff --git a/src/unity_chromiumbookmarks_daemon.py b/src/unity_chromiumbookmarks_daemon.py
index 4cf32c3..64edc39 100644
--- a/src/unity_chromiumbookmarks_daemon.py
+++ b/src/unity_chromiumbookmarks_daemon.py
@@ -134,46 +134,46 @@ def search(search, filters):
     Search for help documents matching the search string
     '''
     results = []
-    global TMP
-    if not os.path.exists(TMP):
-        tmp_folder = tempfile.mkdtemp()
-        TMP = tmp_folder
-    else:
-        tmp_folder = TMP
+#    global TMP
+#    if not os.path.exists(TMP):
+#        tmp_folder = tempfile.mkdtemp()
+#        TMP = tmp_folder
+#    else:
+#        tmp_folder = TMP
 
     for path in get_chromium_profiles():
         user = path.split('/')[-2]
         bookmarks = get_bookmarks_from_chromium(path)
 
-        if DEFAULT_TOP_SITES:
-            thumbnail_path = DEFAULT_TOP_SITES
-        else:
-            thumbnail_path = path.replace('Bookmarks', 'Top Sites')
-        conn = sqlite3.connect(thumbnail_path)
-        connection = conn.cursor()
+#        if DEFAULT_TOP_SITES:
+#            thumbnail_path = DEFAULT_TOP_SITES
+#        else:
+#            thumbnail_path = path.replace('Bookmarks', 'Top Sites')
+#        conn = sqlite3.connect(thumbnail_path)
+#        connection = conn.cursor()
 
         for bookmark in bookmarks:
             # Search bookmark names for matches
             if search.lower() in bookmark[0].lower() or search.lower() in bookmark[1].lower():
-                sqlite_query = '''SELECT thumbnail FROM thumbnails WHERE url = "%s"''' % bookmark[1]
+#                sqlite_query = '''SELECT thumbnail FROM thumbnails WHERE url = "%s"''' % bookmark[1]
                 icon = None
-                connection.execute(sqlite_query)
-                thumb = connection.fetchall()
-                thumbname = bookmark[1].replace('/', '')
-                if thumb:
-                    imageinfo = thumb[0][0]
-                    if imageinfo:
-                        open('%s/%s' % (tmp_folder, thumbname) , 'wb').write(imageinfo)
-                        icon = '%s/%s' % (tmp_folder, thumbname)
-                else:
-                    if os.path.exists('%s/%s' % (tmp_folder, thumbname)):
-                        os.remove('%s/%s' % (tmp_folder, thumbname))
+#                connection.execute(sqlite_query)
+#                thumb = connection.fetchall()
+#                thumbname = bookmark[1].replace('/', '')
+#                if thumb:
+#                    imageinfo = thumb[0][0]
+#                    if imageinfo:
+#                        open('%s/%s' % (tmp_folder, thumbname) , 'wb').write(imageinfo)
+#                        icon = '%s/%s' % (tmp_folder, thumbname)
+#                else:
+#                    if os.path.exists('%s/%s' % (tmp_folder, thumbname)):
+#                        os.remove('%s/%s' % (tmp_folder, thumbname))
                 results.append({'uri': bookmark[1],
                                 'icon': icon,
                                 'category': 0,
                                 'title': bookmark[0],
                                 'user':user})
-        connection.close()
+#        connection.close()
     return results
 
 
-- 
2.55.0

